packages/components/src/tooltip/index.tsx

```diff
@@ -96,9 +98,36 @@ function UnconnectedTooltip(
 	} );
 
 	if ( isNestedInTooltip ) {
+		// Avoid passing certain props added by the useContextSystem hook when
+		// cloning children, as they are not intended for this scenario.
+		const clonedProps = removeExtraPropsAddedByContext(
+			restProps,
+			'Tooltip'
+		);
+		const chainEvent =
+			( eventName: keyof typeof clonedProps ) =>
+			( ...args: unknown[] ): void => {
+				children.props[ eventName ]?.( ...args );
+				clonedProps[ eventName ]?.( ...args );
+			};
 		return isOnlyChild
 			? cloneElement( children, {
-					...restProps,
+					...removeExtraPropsAddedByContext( clonedProps, 'Tooltip' ),
+					// Merge incoming classname with existing classname.
+					className: classNames(
+						children.props.className,
+						clonedProps.className
+					),
+					// Merge incoming inline styles with existing styles.
+					style: {
+						...children.props.style,
+						...clonedProps.style,
+					},
+					// Chain event listeners by calling both the existing and the
+					// incoming callback.
+					onClick: chainEvent( 'onClick' ),
+					onMouseEnter: chainEvent( 'onMouseEnter' ),
+					onBlur: chainEvent( 'onBlur' ),
```

Comment:
@diegohaz I'd like to get a check from your side that this is a good approach — basically making sure that, when we clone `children` in inner nested tooltips (ie. the tooltip part gets ignored), we forward the correct props to the anchor.

I started by merging `className` and `styles`, then I realized that I needed to remove a few extra things added automatically by the context system (see the `removeExtraPropsAddedByContext` function). Finally, I had to chain events, and for now I chose to chain the ones that seem essential to keep the tooltip functioning correctly.

Please let me know if this approach makes sense to you, or if you have suggestions on how to improve it 🙏